src/main/java/com/vmware/vipclient/i18n/base/instances/TranslationMessage.java

```diff
@@ -223,6 +228,47 @@ public boolean postString(final Locale locale, final String component,
 		return cs.getComponentTranslation();
 	}
 
+
+	/**
+	 * get multiple component's translations from VIP server
+	 *
+	 * @param locales
+	 *            language tags to get the translations of them
+	 * @param components
+	 *            names of the components
+	 * @return a map contains all translations of the components of specified locales
+	 */
+	public Map<Locale, Map<String, Map<String, String>>> getStrings(final Set<Locale> locales,
+			final Set<String> components) {
+		logger.info("Start to execute TranslationMessage.getStrings of multiple components of multiple locales.");
+
+		Map<Locale, Map<String, Map<String, String>>> retMap = new HashMap<>();
+		if (null == locales || locales.isEmpty() || null == components || components.isEmpty()) {
+			logger.error(ConstantsMsg.WRONG_PARAMETER + "locales: {}, components: {}.", locales, components);
+			return retMap;
+		}
+
+
+		try {
+			Set<String> convertedLocales = new HashSet<>();
```

Comment:
can move this convert logic into ComponentsService so that the API layer's logic looks more simple?